### ISSUE TITLE
TypeScript abstract method return JSDoc

### ISSUE THREAD

**Issue by RecuencoJones** [buzinas/tslint-eslint-rules#227]
### Bug Report

- __TSLint version__: 5.2.0
- __TypeScript version__: 2.2.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
abstract class Foo { 
  /**
   * @abstract
   * @returns {string} some string.
   */
  public abstract foo(): string;
}
```

with `tslint.json` configuration:

```js
module.exports = {
  rules: {
    ...
    'valid-jsdoc': [true, {
      prefer: {
        return: 'returns'
      },
      preferType: {
        object: 'Object',
        array: 'Array',
        function: 'Function'
      },
      requireReturn: false,
      requireParamDescription: true,
      requireReturnDescription: true
    }],
    ...
  }
};
```

#### Actual behavior

```
ERROR: Foo.ts[1, 21]: unexpected @returns tag; function has no return statement
```

#### Expected behavior

Clean output

**Comment by RecuencoJones** [comment]
Could be fixed by something like the following:

__src/rules/validJsdocRule.js:335__
```js
...
case 'return':
case 'returns':
    hasReturns = true;

    let isAbstract = fn.node.modifiers.some((modifier) => modifier.kind === ts.SyntaxKind.AbstractKeyword);

    if (!isAbstract && !OPTIONS.requireReturn && !fn.returnPresent && tag.type.name !== 'void' && tag.type.name !== 'undefined') {
        this.addFailure(this.createFailure(start, width, Rule.FAILURE_STRING.unexpectedTag(tag.title)));
    }
    else {
        if (!tag.type && OPTIONS.requireReturnType) {
            this.addFailure(this.createFailure(start, width, Rule.FAILURE_STRING.missingReturnType));
        }
        if (!this.isValidReturnType(tag) && !tag.description && OPTIONS.requireReturnDescription) {
            this.addFailure(this.createFailure(start, width, Rule.FAILURE_STRING.missingReturnDescription));
        }
    }
    break;
...
```

**Comment by jmlopez-rod** [comment]
I like your solution. Could you submit a PR?

**Comment by RecuencoJones** [comment]
Sure @jmlopez-rod, I'm a little bit busy rn, will try to create it ASAP
